### PR TITLE
feat(dotnet): template + semver regex

### PR DESCRIPTION
**Pull request by lnu**
### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes/features)

### Description

WIP  
New regex to extract rc part + template 

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/segment_dotnet.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ func (d *dotnet) init(props *properties, env environmentInfo) {
 			{
 				executable: "dotnet",
 				args:       []string{"--version"},
-				regex:      `(?:(?P<version>((?P<major>[0-9]+).(?P<minor>[0-9]+).(?:\d{2})(?P<patch>[0-9]{1}))))`,
+				regex:      `(?P<version>(?P<major>[0-9]+).(?P<minor>[0-9]+).(?P<patch>[0-9]+)(?:-(?P<rc>rc\.[a-zA-Z0-9.]*))?)`,
```
Comment (lnu) [inline src/segment_dotnet.go]: could be something else than RC. Have to check for "real" semver regex(if one exists).  
https://gist.github.com/jhorsman/62eeea161a13b80e39f5249281e17c39

**Comment by lnu** [comment]
sample:
```
{
  "background": "#512BD4", // .NET purple from https://dot.net
  "foreground": "#111111",
  "powerline_symbol": "",
  "properties": {
    "display_version": true,
    "prefix": " \uE77F ",
    "enable_hyperlink": true,
    "template":"{{.Major}}.{{.Minor}}.{{.Patch}}{{if .Prerelease }}.{{ substr 0 4 .Prerelease }}{{end}}",
    "version_url_template":"[{{.Major}}.{{.Minor}}.{{.Patch}}{{if .Prerelease }}.{{ substr 0 4 .Prerelease }}{{end}}](https://github.com/dotnet/core/blob/master/release-notes/{{.Major}}.{{.Minor}}/{{if .Prerelease }}preview/{{.Major}}.{{.Minor}}.0-{{ substr 0 4 .Prerelease }}.md{{else}}{{.Major}}.{{.Minor}}.{{ substr 0 1 .Patch}}/{{.Major}}.{{.Minor}}.{{ substr 0 1 .Patch}}.md{{end}})"
  },
  "style": "powerline",
  "type": "dotnet"
},
```

File: src/segment_dotnet.go (hunk 1 above)
Comment (JanDeDobbeleer) [reply src/segment_dotnet.go]: agreed, could be anything

Comment (lnu) [reply src/segment_dotnet.go]: I used the one from semver.org in the end. Looks good.

Comment (lnu) [reply src/segment_dotnet.go]: https://regex101.com/r/q9rQn3/1

**Comment by lnu** [comment]
I'm not sure what's best: give full flexibility with templates for both text and hyperlink or have a predefined one. It means we have to update all language segments to have a predefined url in the template format.

File: src/properties.go [hunk 2]
```diff
@@ -29,6 +29,8 @@ const (
 	AlwaysEnabled Property = "always_enabled"
 	// SegmentTemplate is the template to use to render the information
 	SegmentTemplate Property = "template"
+	// VersionUrlTemplte i the template to use when building language segment hyperlink
```
Comment (JanDeDobbeleer) [inline src/properties.go]: typo ;-)

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]
1 typo, I would ship it like this for now. Well done.

File: src/properties.go (hunk 2 above)
Comment (lnu) [reply src/properties.go]: that damned butterfly keyboard :sweat_smile:

**Comment by lnu** [comment]
> 1 typo, I would ship it like this for now. Well done.

let me update the doc

**Pull request merged**